spec/services/charges/amount_details/range_graduated_service_spec.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Charges::AmountDetails::RangeGraduatedService, type: :service do
+  subject(:service) { described_class.new(range:, total_units:) }
+  let(:total_units) { 15 }
+  let(:range) do
+    {
+      from_value: 0,
+      to_value: 10,
+      per_unit_amount: '10',
+      flat_amount: '2',
+    }
+  end
+
+  it "returns expected amount details" do
```

Comment:
Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols. (https://rubystyle.guide#consistent-string-literals)